### PR TITLE
differentiate replicated and native helm documentation

### PR DESCRIPTION
**Pull request by morningvera**
This PR adds a little more detail to the native helm docs, renames the old helm workflow to `Replicated Helm` and modifies some existing docs to point to the native helm docs.

I think I caught the places that needed to be updated for native helm but it would be super helpful to get another set of eyes to see if I missed anything. Let me know if anything needs more detail. Thanks!

Note: this shouldn't be merged until we decide to make native Helm GA

### CODE REVIEW

**Comment by JustinN-Replicated** [comment]
I think we will need to adjust the naming of the titles as well, so for example: /vendor/replicated-helm/using-helm-charts/

In the nav bar on the left, it's still listed as 'Helm Charts' and 'Native Helm Charts'

Maybe we need to adjust to something like 'Replicated Helm Deploy' and 'Native Helm Deploy' or something. That likely isn't the best phrasing, but wanted to get the idea out there to consider. Ultimately, it feels like we need to differentiate when replicated and when native helm is doing the install; with the titles 'helm charts' and 'native helm charts' this may be too ambiguous.

**Comment by jonquil2002** [comment]
I agree with Justin's line of thought. Further, I'm also having problems with branding the original deployment as "Replicated Helm Charts" or "Replicated Helm Deployment" because it sounds like our own version of Helm charts.

Perhaps "Deploying Helm as a Replicated Application" and "Deploying Helm using Hooks" conveys the difference more clearly. It would also be good to make that a bit clearer in the topic content as well if you can, and clarify those references to "Replicated Helm".

File: content/reference/v1beta1/helmchart.md [hunk 1]
```diff
@@ -6,10 +6,12 @@ description: "HelmChart defines an instance of a chart"
 weight: 6
 ---
 
-A KOTS HelmChart custom resource enables KOTS to process and [deploy Helm charts as part of a Vendor distributed application](/vendor/helm/using-helm-charts/). 
+A KOTS HelmChart custom resource enables KOTS to process and deploy Helm charts as part of a Vendor distributed application [using the Replicated Helm workflow for existing applications](/vendor/replicated-helm/using-helm-charts/) or [the native Helm workflow for new applications](/vendor/native-helm/using-native-helm-charts/).
```
Comment (jonquil2002) [inline content/reference/v1beta1/helmchart.md:9]: s/ Replicated Helm workflow / Replicated Helm deployment workflow

File: content/vendor/replicated-helm/using-helm-charts.md [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+  - /vendor/replicated-helm/
 weight: 20500
 ---
 
-Helm is a popular package manager for Kubernetes applications. 
-Replicated KOTS supports delivering Helm charts as an enterprise application, or including Helm charts as components of an application. 
-A KOTS application can support more than one Helm chart, and can support more than a single instance of any Helm chart.
+Helm is a popular package manager for Kubernetes applications. Replicated KOTS supports delivering Helm charts as an enterprise application, or including Helm charts as components of an application. A KOTS application can support more than one Helm chart, and can support more than a single instance of any Helm chart.
+
+> The Replicated Helm workflow uses KOTS to render and deploy Helm charts. For new applications, the [native Helm workflow](/vendor/native-helm/) is the preferred method because it supports more features of Helm, such as hooks and weights.
```
Comment (jonquil2002) [inline content/vendor/replicated-helm/using-helm-charts.md:12]: s/ The Replicated Helm workflow / The Replicated deployment workflow of Helm

s/ the [native Helm workflow] / the [native Helm deployment workflow]

File: content/vendor/native-helm/using-native-helm-charts.md [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
+Replicated KOTS already supports delivering [Helm charts](/vendor/replicated-helm/using-helm-charts/) as components of an application but until now, it did not support certain Helm features. With the native Helm installation, you can exercise more control over chart deployment via Helm hooks and weights.
 
 The native Helm chart support currently has the following limitations:
 * Only available for Helm V3.
 * Only supported for new charts added to an existing application or a new application with new charts.
-* Not yet supported on existing charts deployed on existing applications.
+* Not supported on existing charts deployed on existing applications.
 * The pre-rollback, post-rollback, and test hooks are not supported.
 * Hook weights below -9999. All hook weights must be set to a value above -9999 to ensure the Replicated image pull secret is deployed before any resources are pulled.
 * Not supported with the [GitOps Workflow](/kotsadm/gitops/).
```
Comment (JustinN-Replicated) [inline content/vendor/native-helm/using-native-helm-charts.md:16]: * __Not supported with the [GitOps Workflow](/kotsadm/gitops/).__

**Comment by JustinN-Replicated** [comment]
Considering the last changes, maybe the highest level of organization is just 'helm charts' and we cover native helm and replicated helm as sub-topics within the higher level topic. This still looks like it is organized oddly to me (may just be a me thing though)

image

File: content/vendor/native-helm/using-native-helm-charts.md [hunk 4]
```diff
@@ -1,24 +1,27 @@
...
 weight: 70001
-isBeta: true
 ---
 
-Replicated KOTS already supports delivering [Helm charts](/vendor/helm/using-helm-charts/) as components of an application but until now, it did not support certain Helm features. With the new native Helm workflow, you can now exercise more control over chart deployment via Helm hooks and weights. Currently, this is a beta feature and is not recommended for production.
+Replicated KOTS already supports delivering [Helm charts](/vendor/replicated-helm/using-helm-charts/) as components of an application but until now, it did not support certain Helm features. With the native Helm installation, you can exercise more control over chart deployment via Helm hooks and weights.
 
 The native Helm chart support currently has the following limitations:
 * Only available for Helm V3.
 * Only supported for new charts added to an existing application or a new application with new charts.
```
Comment (jeffreygolden) [inline content/vendor/native-helm/using-native-helm-charts.md:12]:
```suggestion
* Only supported for new installations.
```

Suggesting this replaces both line 12 and 13.

File: content/reference/v1beta1/helmchart.md [hunk 5]
```diff
@@ -30,6 +32,9 @@ spec:
   # helmVersion identifies the Helm Version used to render the Chart. Default is v2.
   helmVersion: v2
 
+  # useHelmInstall identifies whether this Helm chart will use the Replicated Helm installation (false) or native Helm installation (true). Default is false.
+  useHelmInstall: true
```
Comment (jeffreygolden) [inline content/reference/v1beta1/helmchart.md:36]: We might have to rethink the wording here.  useHelmInstall: true is not compatible with helmVersion: v2.  Native helm must use helmVersion: v3.

**Comment by JustinN-Replicated** [comment]
@jonquil2002 interested to see what you think of the new organization; I think it is a bit more clear / linear to follow. Only thing I could consider is maybe indenting the pages under the two install sections so it is clear that they are separate but both children of the main 'helm chart' heading. WDYT?

**Comment by jonquil2002** [comment]
@morningvera @JustinN-Replicated 

I'd like to see some subheadings and and some title renaming to make this clearer. Something like:

Overview
Installing with Native Helm
          Native Helm Install Order
Installing with Replicated KOTS
          Replicated Helm Processing
         <additional Replicated Helm-specific topics>
Optional Charts
Optional Value Keys
Helm Airgap Builder

Is this TOC subsuming possible?

Also, in the topics that apply to both flavors of Helm, a clarification sentence at the beginning of each topic could also be helpful to vendors. E.g., "This topic applies to native Helm and Replicated Helm installations."

**Comment by JustinN-Replicated** [comment]
Shared the preview with the CSM/TAM/CE teams. Will follow up if they have any feedback; having said that I think we are in good shape so can always consider iterating in the future as well.

**Comment by jonquil2002** [comment]
lgtm

**Review by jonquil2002 — APPROVED** [review]

**Pull request merged**